src/main/java/org/acme/tasks/scanners/SnykAnalysisTask.java

```diff
@@ -136,49 +136,33 @@ public void analyze(final List<Component> components) {
-                                    .header(HttpHeaders.AUTHORIZATION, this.apiToken);
-                            final HttpResponse<JsonNode> jsonResponse = request.asJson();
-                            if (jsonResponse.getStatus() == 200) {
-                                ArrayList<VulnerableSoftware> vsList = handle(component, jsonResponse.getBody().getObject());
-                                vulnerability.setVulnerableSoftware(vsList);
-                                vulnerablityResult.setComponent(component);
-                                vulnerablityResult.setVulnerability(vulnerability);
-                                vulnerablityResult.setIdentity(AnalyzerIdentity.SNYK_ANALYZER);
-                                vulnerabilityResultProducer.sendVulnResultToDT(component.getUuid(), vulnerablityResult);
-
-                            } else {
-                                handleUnexpectedHttpResponse(LOGGER, API_BASE_URL, jsonResponse.getStatus(), jsonResponse.getStatusText());
-                            }
-                            updateComponentAnalysisCache(ComponentAnalysisCache.CacheType.VULNERABILITY, API_BASE_URL, Vulnerability.Source.SNYK.name(), component.getPurl().toString(), Date.from(Instant.now()), component.getCacheResult());
-
-                        } else {
-                            LOGGER.info("Cache is current, apply analysis from cache");
-                            applyAnalysisFromCache(Vulnerability.Source.SNYK, API_BASE_URL, component.getPurl().toString(), component, getAnalyzerIdentity());
+            int numThreads = 10;
+            int trackComponent = 0;
+            try {
+                while (trackComponent < paginatedList.size()) {
+                    for (int i = 0; i < numThreads; i++) {
+                        final List<Component> temp = new ArrayList<>();
+                        int k = 0;
+                        while (trackComponent < paginatedList.size() && k < numThreads) {
+                            temp.add(paginatedList.get(trackComponent));
+                            k += 1;
+                            trackComponent += 1;
+                        }
+                        for (Component component : temp) {
+                            LOGGER.info(component.getName());
                         }
+                        Thread analysisUtil = new Thread(new SnykAnalysisTaskUtil(temp, apiToken));
+                        analysisUtil.start();
+                        analysisUtil.join();
                     }
-
-                    //cache logic end------------------------------------------
-                } catch (UnirestException e) {
-                    handleRequestException(LOGGER, e);
                 }
+                paginatedComponents.nextPage();
+            } catch (InterruptedException ex) {
+                LOGGER.info("Exception occured: " + ex.getMessage());
+                Thread.currentThread().interrupt();
```

Comment:
Not very confident on this approach. Followed suggestion from sonar for fix

Reply:
I ran into this a while ago, too. It looks weird, but apparently it's the correct way to do it. This StackOverflow answer explains it very well: https://stackoverflow.com/a/3976377

Reply:
Closing this pull request as the changes have already gone in with pull request 39. Will address the review comments from niklas in a different branch regarding the interrupException